### PR TITLE
Added Zend Expressive application adapter

### PR DESCRIPTION
**Pull request by asgrim**
As mentioned in discussion, this provides a Zend Expressive application adapter, whereby the included file should return an instance of `\Zend\Expressive\Application` which is executed by `ZendExpressiveApplicationAdapter`.

It's a little rough, comments welcome!

### CODE REVIEW

File: src/AppFile/ZendExpressiveApplicationAdapter.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        if (!file_exists($path)) {
+            throw new Psr7AppFileException('No file found at ' . $path);
+        }
+
+        $this->path = $path;
+    }
+
+    public function handle(RequestInterface $request): ResponseInterface
+    {
+        if (! $request instanceof ServerRequestInterface) {
```
Comment (asgrim) [inline src/AppFile/ZendExpressiveApplicationAdapter.php:30]: This whole thing could be stripped out if the `Psr7App` interface changes to only accept `ServerRequestInterface`, which I think is probably preferable given the intentions...

Comment (ciaranmcnulty) [reply src/AppFile/ZendExpressiveApplicationAdapter.php:30]: Yes that makes perfect sense - care to do as part of this PR? (we're still in dev-master mode so it's the wild west for now)

**Comment by ciaranmcnulty** [comment]
Overall this looks good, you can make the ServerRequestInterface change as part of this PR

I'll separately add an extension point to select which `Psr7App` is used

File: src/AppFile/ZendExpressiveApplicationAdapter.php (hunk 1 above)
Comment (ciaranmcnulty) [inline src/AppFile/ZendExpressiveApplicationAdapter.php]: Can you take a look at #3 please?

**Comment by ciaranmcnulty** [comment]
I've adopted `ServerRequestInterface`, so some of this PR can be simplified

In #4 I've added extension points that would make it easier to plug these classes into the container - it would also make it possible to add them in a separate behat extension if you'd rather maintain separately? 

I'm not sure whether it's best to have a centralised extension for all PSR-7 apps, or split them out?

**Pull request merged**